### PR TITLE
Add Minimax LLM and Chat

### PR DESCRIPTION
**Pull request by Lukalukaxxx**
- Add support for MiniMax LLM
- [Doc](https://api.minimax.chat/document/algorithm-concept?id=6433f37594878d408fc8295d)

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
example notebook would be great!

**Comment by Lukalukaxxx** [comment]
> example notebook would be great!示例笔记本将是伟大的！

@hwchase17  hi, I add the notebook and chat. Please review : )

**Comment by Lukalukaxxx** [comment]
@hwchase17 Kindly ping for review. Thanks.

File: langchain/chat_models/minimax.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    BaseMessage,
+    ChatGeneration,
+    ChatResult,
+    HumanMessage,
+    SystemMessage,
+)
+logger = logging.getLogger(__name__)
+
+@dataclass
+class _ChatHistory:
```
Comment (baskaryan) [inline langchain/chat_models/minimax.py:25]: @Lukalukaxxx what's this for?

Comment (Lukalukaxxx) [reply langchain/chat_models/minimax.py]: @baskaryan  Sorry, this was a piece of unused code. I've already removed it. Please kindly review it again.

> @Lukalukaxxx what's this for?

**Comment by hetaoBackend** [comment]
Hey @Lukalukaxxx, it's great to see the minimax llm model to be integrated with Langchain. However, I didn't notice that this PR is in review, so I created a PR https://github.com/hwchase17/langchain/pull/7645 for the purpose as well. 

After taking a look, I found all the functionality in https://github.com/hwchase17/langchain/pull/7645 has been covered in this PR, but I'd like to discuss whether it would be better if we extract the calling API logic into an independent class, e.g. https://github.com/hwchase17/langchain/pull/7645/files#diff-0a9ce10f1093aede74365e42383221438316c05f618317f1c7b61b1614223ebcR25-R54? 

It would be genuinely appreciated if u could take a look at this comment!

cc: @baskaryan

**Comment by hetaoBackend** [comment]
> Hey @Lukalukaxxx, it's great to see the minimax llm model to be integrated with Langchain. However, I didn't notice that this PR is in review, so I created a PR #7645 for the purpose as well.
> 
> After taking a look, I found all the functionality in #7645 has been covered in this PR, but I'd like to discuss whether it would be better if we extract the calling API logic into an independent class, e.g. https://github.com/hwchase17/langchain/pull/7645/files#diff-0a9ce10f1093aede74365e42383221438316c05f618317f1c7b61b1614223ebcR25-R54?
> 
> It would be genuinely appreciated if u could take a look at this comment!
> 
> cc: @baskaryan

I've created a PR for this purpose, please check it out when u get a chance! Thanks in advance! @Lukalukaxxx https://github.com/Lukalukaxxx/langchain/pull/1

**Comment by xiaoyucunx** [comment]
Hope to go online as soon as possible, thank you

**Comment by leo-gan** [comment]
@Lukalukaxxx Hi , could you, please, resolve the merging issues? After that ping me and I push this PR for the review. Thanks!

**Comment by hetaoBackend** [comment]
> @Lukalukaxxx Hi , could you, please, resolve the merging issues? After that ping me and I push this PR for the review. Thanks!

Hey @leo-gan , since we could not reach out to @Lukalukaxxx, does it make sense that I create a new PR to cover the enhancement and fix the merging conflicts? It would be appreciated if u could give some insights.

**Comment by leo-gan** [comment]
@hetaoBackend Sure, it would be great! Thank you!

**Comment by hetaoBackend** [comment]
> @hetaoBackend Sure, it would be great! Thank you!

You're very welcome, I will create a new PR and keep u posted as things progress.

**Comment by hetaoBackend** [comment]
> @hetaoBackend Sure, it would be great! Thank you!

Hey @leo-gan , I've submitted a PR https://github.com/langchain-ai/langchain/pull/10776 to resolve the merging conflicts. It would be genuinely appreciated if u could help me review it.

**Comment by leo-gan** [comment]
Closing, since it's functionality was moved to #10776

**Pull request closed without merging**